pkg/clustermanager/eksa_mover.go

```diff
@@ -0,0 +1,108 @@
+package clustermanager
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+// MoverOpt allows to customize a Mover on construction.
+type MoverOpt func(*Mover)
+
+// Mover applies the cluster spec to the management cluster and waits
+// until the changes are fully reconciled.
+type Mover struct {
+	log                                        logr.Logger
+	clientFactory                              ClientFactory
+	applyClusterTimeout, waitForFailureMessage time.Duration
+	retryBackOff                               time.Duration
+}
+
+// NewMover builds an Mover.
+func NewMover(log logr.Logger, clientFactory ClientFactory, opts ...MoverOpt) Mover {
+	m := &Mover{
+		log:                   log,
+		clientFactory:         clientFactory,
+		applyClusterTimeout:   applyClusterSpecTimeout,
+		waitForFailureMessage: waitForFailureMessageErrorTimeout,
+		retryBackOff:          retryBackOff,
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return *m
+}
+
+// WithMoverNoTimeouts disables the timeout for all the waits and retries in management upgrader.
+func WithMoverNoTimeouts() MoverOpt {
+	return func(a *Mover) {
+		maxTime := time.Duration(math.MaxInt64)
+		a.applyClusterTimeout = maxTime
+		a.waitForFailureMessage = maxTime
+	}
+}
+
+// WithMoverApplyClusterTimeout allows to configure how long the mover retries
+// to apply the objects in case of failure.
+// Generally only used in tests.
+func WithMoverApplyClusterTimeout(timeout time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.applyClusterTimeout = timeout
+	}
+}
+
+// WithMoverRetryBackOff allows to configure how long the mover waits between requests
+// to update the cluster spec objects and check the status of the Cluster.
+// Generally only used in tests.
+func WithMoverRetryBackOff(backOff time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.retryBackOff = backOff
+	}
+}
+
+// Move applies the cluster's namespace and spec without checking for reconcile conditions.
+func (m Mover) Move(ctx context.Context, spec *cluster.Spec, targetCluster *types.Cluster) error {
+	var client kubernetes.Client
+	m.log.V(3).Info("Applying cluster spec")
```

Comment:
Can we change this to something different from the applier log, like `moving the cluster object` or something?